### PR TITLE
CSS calc(): clarify units usage

### PR DESCRIPTION
**Pull request by sidewayss**
Clarifies these points:
1) Where units are required
2) Where`<number>` is required, and you *must not* specify units
3) Units are for more than just the `<length>` type, they are for all the types listed in the first paragraph of the page except for `<number>` and `<integer>` (I assume that `<integer>` is a subset of `<number>`, and it seems that this page makes that assumption too).

I tried to modify as little as possible, but there might be a better approach to explaining it - maybe a block of examples, good and bad?

### CODE REVIEW

File: files/en-us/web/css/calc/index.md
Comment (dipikabh) [inline files/en-us/web/css/calc/index.md:1]: A few suggestions:

1. Let's introduce the term "operand" in the paragraph after the syntax:

"The calc() function takes a single expression as its parameter, and the expression's result is used as the value for a CSS property. In this expression, operands, which are the values being operated on, can be combined using the operators listed below. When the expression contains multiple operands, standard operator precedence rules are followed to derive the result."

2. Let's also expand the description of the operators. For example:

```
    - `+`
        - : Adds the specified operands.
    - `-`
        - : Subtracts the second operand from the first operand.
 ```

File: files/en-us/web/css/calc/index.md [hunk 1]
```diff
@@ -25,11 +25,11 @@ The `calc()` function takes a single expression as its parameter, with the expre
 - `-`
   - : Subtraction.
 - `*`
-  - : Multiplication. At least one of the arguments must be a {{cssxref("&lt;number&gt;")}}.
+  - : Multiplication. At least one of the operands must be a {{cssxref("&lt;number&gt;")}}.
```
Comment (dipikabh) [inline files/en-us/web/css/calc/index.md:28]:
```suggestion
  - : Multiplies the specified operands. At least one of the operands must be a {{cssxref("&lt;number&gt;")}}.
```

File: files/en-us/web/css/calc/index.md [hunk 2]
```diff
@@ -25,11 +25,11 @@ The `calc()` function takes a single expression as its parameter, with the expre
...
 - `/`
   - : Division. The right-hand side must be a {{cssxref("&lt;number&gt;")}}.
```
Comment (dipikabh) [inline files/en-us/web/css/calc/index.md:30]:
```suggestion
  - : Divides the left-side operand (dividend) by the right-side operand (divisor). The right-hand operand, the divisor, must be a {{cssxref("&lt;number&gt;")}}.
```

File: files/en-us/web/css/calc/index.md [hunk 3]
```diff
@@ -25,11 +25,11 @@ The `calc()` function takes a single expression as its parameter, with the expre
...
 
-The operands in the expression may be any {{cssxref("&lt;length&gt;")}} syntax value. You can use different units for each value in your expression, if you wish. You may also use parentheses to establish computation order when needed.
+All operands must be suffixed with an appropriate units string, except those that "must be a {{cssxref("&lt;number&gt;")}}". You can use different units for each value in your expression, if you wish. You may also use parentheses to establish computation order when needed.
```
Comment (dipikabh) [inline files/en-us/web/css/calc/index.md:32]:
```suggestion
All operands, except of the type {{cssxref("&lt;number&gt;")}}, must be suffixed with an appropriate unit string, such as `px`, `em`, or `%`. You can use a different unit with each each operand in your expression. You may also use parentheses to establish computation order when needed.
```
I think we can move this paragraph up before the description of the operators.

**Review by dipikabh — CHANGES_REQUESTED** [review]
Hello @sidewayss, these are great clarifications. Great job adding them!

There is scope for more improvement (like adding a "Parameters" section as well as the good/bad examples you suggested) but for now we can limit to the changes you've proposed. Let me know what you think of the suggested edits.

**Comment by sidewayss** [comment]
I have just pushed edits based on your suggestions and including my suggestion of links to the glossary for "operand" and "operator".  Is there a more concise, fancy way to do those links? If so, please use it.

File: files/en-us/web/css/calc/index.md [hunk 4]
```diff
@@ -18,18 +18,18 @@ The **`calc()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_F
 width: calc(100% - 80px);
 ```
 
-The `calc()` function takes a single expression as its parameter, with the expression's result used as the value. The expression can be any simple expression combining the following operators, using standard [operator precedence rules](/en-US/docs/Learn/JavaScript/First_steps/Math#operator_precedence):
+The `calc()` function takes a single expression as its parameter, and the expression's result is used as the value for a CSS property. In this expression, the [operands](/en-US/docs/Glossary/Operand) can be combined using the [operators](/en-US/docs/Glossary/Operator) listed below. When the expression contains multiple operands,`calc()` uses the standard [operator precedence rules](/en-US/docs/Learn/JavaScript/First_steps/Math#operator_precedence):
```
Comment (dipikabh) [inline files/en-us/web/css/calc/index.md:21]:
```suggestion
The `calc()` function takes a single expression as its parameter, and the expression's result is used as the value for a CSS property. In this expression, the {{Glossary("operand", "operands")}} can be combined using the {{Glossary("operator", "operators")}} listed below. When the expression contains multiple operands,`calc()` uses the standard [operator precedence rules](/en-US/docs/Learn/JavaScript/First_steps/Math#operator_precedence):
```

**Review by dipikabh — CHANGES_REQUESTED** [review]

**Review by dipikabh — APPROVED** [review]
Thanks for the updates! 👍

**Comment by sidewayss** [comment]
> Thanks for the updates! 👍

You're welcome.

**Pull request merged**